lib/ansible/plugins/loader.py

```diff
@@ -1069,23 +1069,35 @@ def __init__(self, class_name, package, config, subdir, aliases=None, required_b
 
         super(Jinja2Loader, self).__init__(class_name, package, config, subdir, aliases=aliases, required_base_class=required_base_class)
         self._loaded_j2_file_maps = []
+        self._j2_module_cache = {}
 
     def _clear_caches(self):
         super(Jinja2Loader, self)._clear_caches()
         self._loaded_j2_file_maps = []
+        self._j2_module_cache = {}
 
     def find_plugin(self, name, mod_type='', ignore_deprecated=False, check_aliases=False, collection_list=None):
 
-        # TODO: handle collection plugin find, see 'get_with_context'
-        # this can really 'find plugin file'
-        plugin = super(Jinja2Loader, self).find_plugin(name, mod_type=mod_type, ignore_deprecated=ignore_deprecated, check_aliases=check_aliases,
-                                                       collection_list=collection_list)
+        plugin = None
+        names = ['.'.join(cname, name) for cname in collection_list] + [name]
```

Comment:
```suggestion
        names = ['.'.join(cname, name) for cname in collection_list or []] + [name]
```
`collection_list` defaults to `None` (though I don't know if it ever practically uses that default, probably best to handle it)